fix: Apply filenamePattern setting when saving notes

ejester-gusto opened the pull request:
## Summary

Fixes #71 - The `filenamePattern` setting was being ignored when saving notes.

- Notes were always saved with just the title (e.g., `Meeting Name.md`) regardless of the configured pattern
- Now respects patterns like `{date}-{title}` producing `2024-12-30-Meeting Name.md`

## Changes

- Added `DocumentProcessorSettings` interface with `filenamePattern` and `transcriptFilenamePattern`
- Added `generateFilename()` helper method to `DocumentProcessor`
- Updated `prepareNote`, `prepareCombinedNote`, and `prepareTranscript` to use pattern-based filename generation
- Updated `main.ts` to pass filename patterns to `DocumentProcessor`
- Added tests for filename pattern functionality

## Test plan

- [x] Added unit tests for filenamePattern in notes
- [x] Added unit tests for transcriptFilenamePattern in transcripts  
- [x] All 240 tests pass
- [x] Build succeeds
- [ ] Manual testing with various filename patterns

ejester-gusto committed:
fix: Apply filenamePattern setting when saving notes

The filenamePattern setting (e.g., {date}-{title}) was being ignored when
saving notes. Files were always saved with just the title regardless of
the configured pattern.

The issue was in DocumentProcessor which was using sanitizeFilename(title)
directly instead of resolveFilenamePattern() from filenameUtils.

Changes:
- Add DocumentProcessorSettings interface with filenamePattern and
  transcriptFilenamePattern
- Add generateFilename() helper method to DocumentProcessor
- Update prepareNote, prepareCombinedNote, and prepareTranscript to use
  the pattern-based filename generation
- Update main.ts to pass filename patterns to DocumentProcessor
- Add tests for filename pattern functionality

Fixes #71

tomelliot commented on src/services/documentProcessor.ts:
```diff
@@ -1,20 +1,49 @@
+   * @param isTranscript - Whether this is a transcript file
+   * @returns The generated filename with .md extension
+   */
+  private generateFilename(
```
We have `utils/filenameUtils.ts` - that would be a better place for this function. Let's refactor `resolveFilenamePattern` to accept a `doc` and a `pattern`. That will keep the calling functions more DRY and keep things simpler.

There'll be a few places we'll need to propagate this same change (e.g. `computeTranscriptPath`, `computeNotePath`).

tomelliot reviewed (changes requested):
Thanks for catching this and for the PR @ejester-gusto.

If you don't want to make the adjustments you can add me as a collaborator in your fork of the repo and I can make them.

ejester-gusto committed:
refactor: Move filename resolution to filenameUtils

Address review feedback:
- Add resolveDocFilename() to utils/filenameUtils.ts that accepts
  a doc and pattern directly, making calling code more DRY
- Simplify DocumentProcessor to use resolveDocFilename()
- Remove generateFilename() private method from DocumentProcessor
- Add tests for resolveDocFilename()

ejester-gusto commented:
Thanks for the feedback! I've made the requested changes:

- Added `resolveDocFilename(doc, pattern)` to `utils/filenameUtils.ts` - this is a convenience function that extracts title and date from the doc and calls `resolveFilenamePattern`
- Simplified `DocumentProcessor` to use the new function instead of having its own `generateFilename` method
- Added tests for `resolveDocFilename`

I kept `resolveFilenamePattern(pattern, title, noteDate)` as-is since `PathResolver` uses it with pre-extracted title/date values. Let me know if you'd prefer to refactor `PathResolver` as well to use the doc-based function.

tomelliot committed:
refactor: pass GranolaDoc to path resolver methods

tomelliot committed:
refactor: consolidate filename resolution into single function

tomelliot committed:
test: improve mock setup and test isolation documentation

tomelliot committed:
refactor: Move filename pattern logic to PathResolver

tomelliot committed:
docs: Add testing rules for mocking vs real dependencies

tomelliot committed:
docs: Remove legacy migration section and add code patterns guide
